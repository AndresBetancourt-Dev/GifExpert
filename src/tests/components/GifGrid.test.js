import React from "react";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Test of Component : <GifGrid/>", () => {
  const category = "One Punch Man";

  test("GifGrid : Component Mounts Correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBeTruthy();
  });
  test("GifGrid : Component Mounts Correctly With Use Fetch", () => {
    const gifs = [
      { id: "ABC", url: "https://something.com", title: "ABC" },
      { id: "ABCD", url: "https://something2.com", title: "ABCD" },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBeFalsy();
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
