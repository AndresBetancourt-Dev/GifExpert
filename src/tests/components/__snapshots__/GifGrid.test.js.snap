// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Component : <GifGrid/> GifGrid : Component Mounts Correctly 1`] = `
<div
  className="grid animate__animated animate__fadeIn"
>
  <h3>
    One Punch Man
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid "
  />
</div>
`;

exports[`Test of Component : <GifGrid/> GifGrid : Component Mounts Correctly With Use Fetch 1`] = `
<div
  className="grid animate__animated animate__fadeIn"
>
  <h3>
    One Punch Man
  </h3>
  <div
    className="card-grid "
  >
    <GifGridItem
      key="ABC"
      title="ABC"
      url="https://something.com"
    />
    <GifGridItem
      key="ABCD"
      title="ABCD"
      url="https://something2.com"
    />
  </div>
</div>
`;
